### PR TITLE
[TA] 3.1 swagger changes

### PR DESCRIPTION
**Pull request by mssfang**
- Regenerated 3.1 swagger and disabled all failed tests (created an issue for it https://github.com/Azure/azure-sdk-for-java/issues/22208).
- In order to build the package,  added a new class, `HealthcareEntityCategory` to replace the `String` type of property `category` in the `HealthcareEntity`.

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # Release History
 
-## 5.1.0-beta.8 (Unreleased)
-
+## 5.1.0 (Unreleased)
+#### Feature Added
+- Added a new class, `HealthcareEntityCategory` to replace the `String` type of property `category` in the `HealthcareEntity`.
```
Comment (deyaaeldeen) [inline sdk/textanalytics/azure-ai-textanalytics/CHANGELOG.md:5]: Also, can you add that now it targets service v3.1?

**Review by maririos — COMMENTED** [review]
Are you going to change the CI so that it runs with a specific endpoint and API key while the deployment happens in prod regions? or are you going to leave this PR open until then?

Also, by targeting a new service version, do you need to re-record tests? I know in .NET I have to as the endpoint is different

**Review by deyaaeldeen — APPROVED** [review]
Let's get this PR merged into a feature branch instead of master. I left one comment.

**Comment by mssfang** [comment]
Seems like it is better to keep this PR as the feature branch PR. The last PR to merge into master.

(1) Consider this branch as a feature branch.
(2) I will create other PRs that includes the smaller changes and targeting to merge into this feature branch PR.
(You will get an overview changes each time. but feel free to ignore the github notification from this PR until all features/changes are merged into this feature branch)

**Comment by mssfang** [comment]
Since the branch is from my forked repo, so can't add you as the reviewer, I would still need to create a feature branch in this repo but not my forked repo. Closing this PR.

**Comment by mssfang** [comment]
> Are you going to change the CI so that it runs with a specific endpoint and API key while the deployment happens in prod regions? or are you going to leave this PR open until then?
> 
> Also, by targeting a new service version, do you need to re-record tests? I know in .NET I have to as the endpoint is different

Yes.  When a new service version is merged in, we need to re-record the tests. Currently, the PR is still targeting preview-5, the TA service version class should also need to be updated in order to test v3.1.

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java [hunk 2]
```diff
@@ -701,7 +701,7 @@ static AnalyzeHealthcareEntitiesResult getRecognizeHealthcareEntitiesResult1(Str
             IterableStream.of(Collections.emptyList()));
         final HealthcareEntity healthcareEntity3 = new HealthcareEntity();
         HealthcareEntityPropertiesHelper.setText(healthcareEntity3, "progressive");
-        HealthcareEntityPropertiesHelper.setCategory(healthcareEntity3, "ConditionQualifier");
+        HealthcareEntityPropertiesHelper.setCategory(healthcareEntity3, HealthcareEntityCategory.fromString("Course"));
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java:704]: is this a category that is not avialable in the swagger? or why is it different?

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java [hunk 3]
```diff
@@ -738,13 +738,13 @@ static AnalyzeHealthcareEntitiesResult getRecognizeHealthcareEntitiesResult1(Str
...
         final HealthcareEntityRelationRole role1 = new HealthcareEntityRelationRole();
-        HealthcareEntityRelationRolePropertiesHelper.setName(role1, "Qualifier");
+        HealthcareEntityRelationRolePropertiesHelper.setName(role1, "Course");
         HealthcareEntityRelationRolePropertiesHelper.setEntity(role1, healthcareEntity3);
         final HealthcareEntityRelationRole role2 = new HealthcareEntityRelationRole();
         HealthcareEntityRelationRolePropertiesHelper.setName(role2, "Condition");
         HealthcareEntityRelationRolePropertiesHelper.setEntity(role2, healthcareEntity4);
         HealthcareEntityRelationPropertiesHelper.setRelationType(healthcareEntityRelation1,
-            HealthcareEntityRelationType.QUALIFIER_OF_CONDITION);
+            HealthcareEntityRelationType.fromString("CourseOfCondition"));
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java:747]: same question on why the need to create the type...

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java [hunk 4]
```diff
@@ -800,6 +802,7 @@ public void recognizePiiEntitiesDiacriticsNfd(HttpClient httpClient, TextAnalyti
 
     @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
     @MethodSource("com.azure.ai.textanalytics.TestUtils#getTestParameters")
+    @Disabled("아가 means baby but recognized as an person. Previously, this is not recognized as Pii category. https://github.com/Azure/azure-sdk-for-java/issues/22208")
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java:805]: Having these errors means you need to update your tests? or is there any regression or bug in the service?

**Review by maririos — APPROVED** [review]

**Review by kristapratico — APPROVED** [review]

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java (hunk 4 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java:805]: It seems more on the service side regression. Will create a service issue.

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java (hunk 2 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java:704]: The category is not available in the swagger. So we can have it by extending it.

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java (hunk 3 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java:747]: same answer in https://github.com/Azure/azure-sdk-for-java/pull/22174#discussion_r652169498

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java (hunk 2 above)
Comment (maririos) [reply sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TestUtils.java:704]: ohhh we should push either the service team to add it to the swagger, or you can go ahead and create a PR for their swagger that adds the missing types you found

**Pull request merged**